core/src/main/java/org/geysermc/geyser/session/cache/RegistryCache.java

```diff
@@ -149,25 +149,35 @@ public void load(ClientboundRegistryDataPacket packet) {
      * @param reader converts the RegistryEntry NBT into a class file
      * @param <T> the class that represents these entries.
      */
-    private static <T> void register(String registry, Function<RegistryCache, JavaRegistry<T>> localCacheFunction, BiFunction<GeyserSession, RegistryEntry, T> reader) {
-        Key key = MinecraftKey.key(registry);
-        REGISTRIES.put(key, (registryCache, entries) -> {
+    private static <T> void register(String registry, Function<RegistryCache, JavaRegistry<T>> localCacheFunction, Function<RegistryEntryContext, T> reader) {
+        Key registryKey = MinecraftKey.key(registry);
+        REGISTRIES.put(registryKey, (registryCache, entries) -> {
             Map<Key, NbtMap> localRegistry = null;
             JavaRegistry<T> localCache = localCacheFunction.apply(registryCache);
             // Clear each local cache every time a new registry entry is given to us
             // (e.g. proxy server switches)
+
+            // Store each of the entries resource location IDs and their respective network ID,
+            // used for the key mapper that's currently only used by the Enchantment class
+            Map<Key, Integer> entryIdMap = new HashMap<>();
```

Comment:
This can be a Object2IntMap.

Reply:
Seems a bit wasteful to run this when it's rarely used, but for now it's fine. We can come back to it and lazily initialize it.